standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreSchemaInfo.java

```diff
@@ -205,19 +206,32 @@ public boolean isVersionCompatible(String hiveVersion, String dbVersion) {
       return false;
     }
 
-    for (int i = 0; i < dbVerParts.length; i++) {
-      int dbVerPart = Integer.parseInt(dbVerParts[i]);
-      int hiveVerPart = Integer.parseInt(hiveVerParts[i]);
-      if (dbVerPart > hiveVerPart) {
-        return true;
-      } else if (dbVerPart < hiveVerPart) {
-        return false;
-      } else {
-        continue; // compare next part
+    int i = 0;
```

Comment:
This makes me nervous 😄 
Why do we have to move around the declaration of `i`?

Maybe the code would be easier to read this way:
```
hiveVerParts = hiveVersion.split("\\.|-");
dbVerParts = dbVersion.split("\\.|-");
for (int i; i < Math.min(hiveVerParts.length, dbVerParts.length); i++) {
  int compare = compareVersion(dbVerParts[i], hiveVerParts[i]);
  if (compare != 0) {
    return compare > 0;
  }
}

// This is always false, as we already checked that hiveVersion and dbVersions are different 
return hiveVerParts.length == dbVerParts.length;
```

Reply:
If `i` reaches the end of `dbVerParts`, means that the hive version is behind the db version, which supposed to be compatiable. For example,
dbVerParts is `4`, `0`, `0`
hiveVerParts is `4`, `0`, `0`, `alpha`, `1`

the `hiveVerParts.length == dbVerParts.length;` returns false in this case.

Reply:
Then this should be always `true`?
Or
`hiveVerParts.length > dbVerParts.length`?

Reply:
`hiveVerParts.length > dbVerParts.length` makes sense, I will fix that. Thank you for your suggestion!